### PR TITLE
Issue #7659: Update doc for SimplifyBooleanReturn

### PR DESCRIPTION
**Pull request by wilcoln**
# Fix for issue #7659  
example

## Ouput of default example
```
$ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>
    <module name="TreeWalker">
		<module name="SimplifyBooleanReturn"/>
	</module>
</module>

$ cat Test.java 
public class Test {

 private boolean cond;
 private Foo a;
 private Foo b;


 public boolean check1() {
  if (cond) { // violation, can be simplified
    return true;
  }
  else {
    return false;
  }
 }

 // Ok, simplified version of check1()
 public boolean check2() {
  return cond;
 }

 // violations, can be simplified
 public boolean check3() {
  if (cond == true) { // can be simplified to "if (cond)"
    return false;
  }
  else {
    return true; // can be simplified to "return !cond"
  }
 }

 // Ok, can be simplified but doesn't return a Boolean
 public Foo choose1() {
  if (cond) {
    return a;
  }
  else {
    return b;
  }
 }

 // Ok, simplified version of choose1()
 public Foo choose2()  {
  return cond ? a: b;
 }

}

$ java -jar checkstyle-8.29-all.jar -c config.xml Test.java
Starting audit...
[ERROR] /var/tmp/Test.java:9:3: Conditional logic can be removed. [SimplifyBooleanReturn]
[ERROR] /var/tmp/Test.java:24:3: Conditional logic can be removed. [SimplifyBooleanReturn]
Audit done.
Checkstyle ends with 2 errors.

```

## Ouput of non-default example
This check doesn't provide any configurable property.

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/SimplifyBooleanReturnCheck.java [hunk 1]
```diff
@@ -51,6 +51,41 @@
...
+ *  private boolean cond;
+ *  private Foo a, b;
+ *
+ *  public boolean check1() {
+ *  if (cond)
+ *    return true;
+ *  else
+ *    return false;
+ *  }
+ *  // violation, could be simplified
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/SimplifyBooleanReturnCheck.java]: Please put violation comment on line where violation is reported

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/SimplifyBooleanReturnCheck.java [hunk 2]
```diff
@@ -51,6 +51,41 @@
...
+ *
+ *  public boolean check2() {
+ *   return cond;
+ *  }
+ *  // Ok, simplified version of check1()
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/SimplifyBooleanReturnCheck.java]: Move before method

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/SimplifyBooleanReturnCheck.java [hunk 3]
```diff
@@ -51,6 +51,41 @@
...
+ *
+ *  public Foo choose1() {
+ *  if (cond)
+ *    return a;
+ *  else
+ *    return b;
+ *  }
+ *  // Ok, could be simplified but doesn't return a Boolean
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/SimplifyBooleanReturnCheck.java]: same, move before

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/SimplifyBooleanReturnCheck.java [hunk 4]
```diff
@@ -51,6 +51,41 @@
...
+ *
+ *  public Foo choose2() {
+ *   return cond? a: b;
+ *  }
+ *  // Ok, simplified version of choose1()
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/SimplifyBooleanReturnCheck.java]: same

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/SimplifyBooleanReturnCheck.java [hunk 5]
```diff
@@ -51,6 +51,41 @@
...
+ *
+ *  private boolean cond;
+ *  private Foo a, b;
+ *
+ *  public boolean check1() {
+ *  if (cond)
+ *    return true;
+ *  else
+ *    return false;
+ *  }
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/SimplifyBooleanReturnCheck.java]: Indentation is missing here and in other if-else example below.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/SimplifyBooleanReturnCheck.java [hunk 6]
```diff
@@ -51,6 +51,41 @@
...
  * &lt;module name=&quot;SimplifyBooleanReturn&quot;/&gt;
  * </pre>
+ * <p>Example: </p>
+ * <pre>
+ * public class Test {
+ *
+ *  private boolean cond;
+ *  private Foo a, b;
+ *
+ *  public boolean check1() {
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/SimplifyBooleanReturnCheck.java]: Please add one more example of violation (some other case, can be checked in test sources for this check)

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/SimplifyBooleanReturnCheck.java (hunk 1 above)
Comment (strkkk) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/SimplifyBooleanReturnCheck.java]: Please do not resolve comments, reviewers should do it.

In this case comment is not on the line where violation occurs, but before method

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/SimplifyBooleanReturnCheck.java [hunk 7]
```diff
@@ -51,6 +51,52 @@
...
+ *     return false;
+ *  }
+ *
+ *  // Ok, simplified version of check1()
+ *  public boolean check2() {
+ *   return cond;
+ *  }
+ *
+ *  // violations, can be simplified
+ *  public boolean check3(){
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/SimplifyBooleanReturnCheck.java]: space missing between ) and {

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/SimplifyBooleanReturnCheck.java [hunk 8]
```diff
@@ -51,6 +51,52 @@
...
+ *     return true; // can be simplified to "return !cond"
+ *   }
+ *  }
+ *
+ *  // Ok, can be simplified but doesn't return a Boolean
+ *  public Foo choose1() {
+ *   if (cond)
+ *     return a;
+ *   else
+ *     return b;
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/SimplifyBooleanReturnCheck.java]: please add curly braces to if-else blocks to make all examples look in same way

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/SimplifyBooleanReturnCheck.java [hunk 9]
```diff
@@ -51,6 +51,52 @@
...
  * </pre>
+ * <p>Example: </p>
+ * <pre>
+ * public class Test {
+ *
+ *  private boolean cond;
+ *  private Foo a;
+ *  private Foo b;
+ *
+ *  // violation, can be simplified
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/SimplifyBooleanReturnCheck.java]: Comment is not on line where violation is

**Comment by strkkk** [comment]
@wilcoln Please do not resolve review items, reviewers do it if item is corrected.

**Comment by wilcoln** [comment]
> @wilcoln Please do not resolve review items, reviewers do it if item is corrected.

Ok noted, sorry

**Comment by wilcoln** [comment]
@strkkk, a review please.

**Pull request merged**